lib/v8.js

```diff
@@ -397,6 +406,46 @@ function deserialize(buffer) {
   return der.readValue();
 }
 
+/**
+ * @param {{
+ *   duration: number,
+ *   signal?: AbortSignal
+ * }} options
+ */
+function collectGCProfile(options) {
+  validateObject(options, 'options');
+  validateNumber(options.duration, 'options.duration', 1);
+  if (options.signal) {
+    validateAbortSignal(options.signal, 'options.signal');
+  }
+  return new Promise((resolve, reject) => {
+    let profiler;
+    let timer;
+    const onAbortListener = () => {
+      timer && clearTimeout(timer);
+      profiler && profiler.stop();
+      reject(new AbortError(undefined, { cause: options.signal.reason }));
+    };
+    if (options.signal) {
+      const { signal } = options;
+      if (signal.aborted) {
+        onAbortListener();
+        return;
+      }
+      signal.addEventListener('abort', onAbortListener, { once: true });
+    }
+    profiler = new binding.GCProfiler();
+    profiler.start();
+    timer = setUnrefTimeout(() => {
+      const data = profiler.stop();
+      resolve(JSONParse(data));
+      if (options.signal) {
+        options.signal.removeEventListener('abort', onAbortListener);
+      }
+    }, options.duration);
```

Comment:
You can save all this logic here if you expose a wrapper around `profiler.stop()`, which seems a bit more flexible in general for consumers – I wonder if there’s a specific reason for preferring a timer?